examples/step-71/step-71.cc

```diff
@@ -0,0 +1,720 @@
+    std::vector<types::global_dof_index> joint_dof_indices;
+  };
+
+
+
+  struct CopyData
+  {
+    FullMatrix<double>                   cell_matrix;
+    Vector<double>                       cell_rhs;
+    std::vector<types::global_dof_index> local_dof_indices;
+    std::vector<CopyDataFace>            face_data;
+
+    template <class Iterator>
+    void reinit(const Iterator &cell, unsigned int dofs_per_cell)
+    {
+      cell_matrix.reinit(dofs_per_cell, dofs_per_cell);
+      cell_rhs.reinit(dofs_per_cell);
+
+      local_dof_indices.resize(dofs_per_cell);
+      cell->get_dof_indices(local_dof_indices);
+    }
+  };
+
+
+
+  template <int dim>
+  void BiharmonicProblem<dim>::assemble_system()
+  {
+    using Iterator = decltype(dof_handler.begin_active());
+    const ExactSolution::RightHandSide<dim> right_hand_side;
+
+    auto cell_worker = [&](const Iterator &  cell,
+                           ScratchData<dim> &scratch_data,
+                           CopyData &        copy_data) {
+      const unsigned int n_dofs = scratch_data.fe_values.get_fe().dofs_per_cell;
+      copy_data.reinit(cell, n_dofs);
+      scratch_data.fe_values.reinit(cell);
+
+      const auto &q_points = scratch_data.fe_values.get_quadrature_points();
+
+      const FEValues<dim> &      fe_v = scratch_data.fe_values;
+      const std::vector<double> &JxW  = fe_v.get_JxW_values();
+
+      const double nu = 1.0;
+
+      for (unsigned int point = 0; point < fe_v.n_quadrature_points; ++point)
+        {
+          for (unsigned int i = 0; i < n_dofs; ++i)
+            {
+              for (unsigned int j = 0; j < n_dofs; ++j)
+                {
+                  // \int_Z \nu \nabla^2 u \cdot \nabla^2 v \, dx.
+                  copy_data.cell_matrix(i, j) +=
+                    nu *
+                    scalar_product(fe_v.shape_hessian(i, point),
+                                   fe_v.shape_hessian(j, point)) *
+                    JxW[point]; // dx
+                }
+
+              copy_data.cell_rhs(i) += fe_v.shape_value(i, point) *
+                                       right_hand_side.value(q_points[point]) *
+                                       JxW[point]; // dx
+            }
+        }
+    };
+
+
+    auto face_worker = [&](const Iterator &    cell,
+                           const unsigned int &f,
+                           const unsigned int &sf,
+                           const Iterator &    ncell,
+                           const unsigned int &nf,
+                           const unsigned int &nsf,
+                           ScratchData<dim> &  scratch_data,
+                           CopyData &          copy_data) {
+      FEInterfaceValues<dim> &fe_i = scratch_data.fe_interface_values;
+      fe_i.reinit(cell, f, sf, ncell, nf, nsf);
+      const auto &q_points = fe_i.get_quadrature_points();
+
+      copy_data.face_data.emplace_back();
+      CopyDataFace &copy_data_face = copy_data.face_data.back();
```

Comment:
@tjhei -- you're adding a face data object to the end of the vector here (and you do the same for the boundary worker), and then their contents are copied in the `copier` function. But you never actually remove them from the vector again. So I have a suspicion that you're actually copying the same data over and over into the matrix. Do you agree that this looks like a bug?